### DISCUSSION TITLE
Jump-In related Scripting API (1.10.4)

### DISCUSSION THREAD

**Discussion by macmarrum** [freeplane/freeplane#588] [General]
Further to 
* https://github.com/freeplane/freeplane/discussions/560#discussioncomment-3390915
* https://github.com/freeplane/freeplane/discussions/560#discussioncomment-3451052

I've been thinking about `NodeRO#getViewSide()`. The name suggests that `side` is associated with a view. It's unclear what view is meant. 
1. If view is meant as "what is displayed", `viewSide` fails to reflect it. Nodes don't have to be displayed based on their `viewSide`. The information about the side where nodes are displayed is delivered by `NodeRO#isLeft()` and `NodeRO#isLeftOnView(NodeRO)`. BTW, the latter has the same issue: little clarity about what is meant by "on view".
2. If view is meant as "jump-in view", `viewSide` delivers more then that, because it covers also the "jump-out view", i.e. for the main root.

Taking these into consideration and keeping in mind that the meaning of `side` is "node's display side when parent is root", I believe the name should rather be `side`, i.e. `NodeRO#getSide()`.
The enum could be named `Side` or `NodeSide`.

As to the meaning of `ifLeftOnView(...)`, I'd put it as "is displayed on the left when the following node is root". I vote for `isLeftFor(NodeRO)` or `isLeftWhenRoot(NodeRO)`.

**Comment by dpolivaev** [comment]
For me, there is one and only node map root. It does not change when you jump in.
Instead, some non-root node is selected as a root of the view. Therefore I call it "view Root".
The related enum is called "ViewSide" because it defines the side of any node at any view with any given view root.
The suggested name "isLeftFor" does not seem clear enough.
The suggested name "isLeftWhenRoot" seems to be confusing because jumping in you do not change the map root, only the view root.
Therefore I like the chosen name "ifLeftOnView" better than the other suggested alternatives.

**Reply by macmarrum** [reply]
> For me, there is one and only node map root. It does not change when you jump in. 
Instead, some non-root node is selected as a root of the view. Therefore I call it "view Root". 

"view Root" is a good name :+1:
I think also _the one and only map root_ can be called "view Root" when in jump-out view.
The API methods seem to agree
- `ControllerRO#getViewRoot`
- `NodeRO#isLeftOnView`
- `NodeRO#isVisibleOnView`

> The related enum is called "ViewSide" because it defines the side of any node at any view with any given view root. 

Let's keep in mind the meaning of the property, i.e. "node's display side when parent is viewRoot".
Therefore I have trouble understanding what value "View" brings in "ViewSide". To me "View" brings more confusion because it suggest this is the side where the node is viewed/displayed, and we'd be better off with just "Side".

> The suggested name "isLeftFor" does not seem clear enough. 

True. A clear name would be "isDisplayedOnTheLeftWhenTheFollowingNodeIsViewRoot", but that's a bit long for a public api method.

> The suggested name "isLeftWhenRoot" seems to be confusing because jumping in you do not change the map root, only the view root. 

True. Let's substitute root with viewRoot then => "isLeftWhenViewRoot"

> Therefore I like the chosen name "ifLeftOnView" better than the other suggested alternatives.

Now there are more alternatives to consider :smiley:

**Reply by dpolivaev** [reply]
What do you think about
`ViewSide` -> `Side`
`isLeftOnView` -> `isLeftOnViewsWithRoot`

**Reply by macmarrum** [reply]
Great. I like it :+1: 

I understand that 
`node.viewSide` will change to `node.side`, 
and `isVisibleOnView` to `isVisibleOnViewsWithRoot`

**Reply by dpolivaev** [reply]
I would change `node.viewSide` to `node.preferredSide`

**Reply by macmarrum** [reply]
> I would change `node.viewSide` to `node.preferredSide`

The information that the property will carry is "node's display side when node's parent is viewRoot".
The "preference" aspect is missing here, in my opinion, therefore it makes little sense to use "preferred" as part of the property name.

Would you care to explain why `node.preferredSide` instead of `node.side`?

**Reply by dpolivaev** [reply]
Because if the node is not a direct child of the view root node, this setting is ignored.

**Reply by macmarrum** [reply]
I feel like the explanation provides very little added value. It just restates what was said before.

If you are worried that the property will be misunderstood, it might make sense to make the name more related to what it does, e.g.

- `sideWhenParentIsViewRoot`
- `sideWhenChildOfViewRoot`

But adding "preferred" to the name fails to do that, in my opinion.

After Cambridge Dictionary

> ### [preferred](https://dictionary.cambridge.org/dictionary/english/preferred)
> 
> [liked](https://dictionary.cambridge.org/dictionary/english/like) or [wanted](https://dictionary.cambridge.org/dictionary/english/wanted) more than anything [else](https://dictionary.cambridge.org/dictionary/english/else):
> - _The [earlier](https://dictionary.cambridge.org/dictionary/english/earlier) [train](https://dictionary.cambridge.org/dictionary/english/train) would be my preferred [option](https://dictionary.cambridge.org/dictionary/english/option)._
> 
> Synonym
[favourite ](https://dictionary.cambridge.org/dictionary/english/favourite)UK

Java doc for `node.getSide()` can explain the meaning of the information and point to `isLeftOnViewsWithRoot(...)` as the method to establish the display side of a node.

**Reply by dpolivaev** [reply]
Yes, `viewSide` and `preferredSide` are not self explanatory. But just `side` is likely to be misunderstood and taken instead of `isLeft`, and the other alternatives are too long IMHO. Therefore I see only these two options. Which of them do you like more? Or which one you would favor? It should be not just `side`but also not 5-6 words long as your counterexamples.

**Reply by macmarrum** [reply]
I still prefer something else to `viewSide` and `preferredSide`.
How about `levelOneSide`?
It's the side when a node is on level 1 in a view.

**Reply by dpolivaev** [reply]
`nearRootSide`? `sideNearRoot`? `sideAtRoot`?

**Reply by dpolivaev** [reply]
Probably, the last one

**Reply by macmarrum** [reply]
Yes. That's the one. `sideAtRoot` :+1:

**Reply by dpolivaev** [reply]
Preview 1.10.4_12
